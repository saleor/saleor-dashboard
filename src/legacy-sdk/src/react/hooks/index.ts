export * from "./auth";
export * from "./saleorConfig";
export * from "./user";
