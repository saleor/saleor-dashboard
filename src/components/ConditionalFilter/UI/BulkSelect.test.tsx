import { fireEvent, render } from "@testing-library/react";
import * as React from "react";

import BulkSelect, { BulkSelectProps } from "./BulkSelect";

const intersectionObserverMock = () => ({
  observe: () => null,
  unobserve: () => null,
});

window.IntersectionObserver = jest.fn().mockImplementation(intersectionObserverMock);

describe("ConditionalFilter / UI / BulkSelect", () => {
  const defaultProps: BulkSelectProps = {
    selected: {
      options: [],
      loading: false,
      conditionValue: {
        type: "bulkselect",
        label: "in",
        value: "input-1",
      },
      value: [],
    },
    onBlur: jest.fn(),
    onFocus: jest.fn(),
    onOptionsChange: jest.fn(),
    dataTestId: "right-0",
    error: false,
    helperText: "Helper text",
    disabled: false,
  };

  it("renders BulkSelect", () => {
    // Arrange & Act
    const { baseElement } = render(<BulkSelect {...defaultProps} />);

    // Assert
    expect(baseElement).toBeInTheDocument();
  });

  it("renders options inside the input", () => {
    // Arrange
    const element = render(
      <BulkSelect
        {...defaultProps}
        selected={{
          ...defaultProps.selected,
          value: [
            {
              label: "value1",
              value: "value1",
              slug: "value1",
            },
          ],
        }}
      />,
    );

    // Act & Assert
    expect(element.getByText("value1")).toBeInTheDocument();
  });

  it("renders options inside the input with more than 15 characters", () => {
    const element = render(
      <BulkSelect
        {...defaultProps}
        selected={{
          ...defaultProps.selected,
          value: [
            {
              label: "verylongvalueofanid",
              value: "verylongvalueofanid",
              slug: "verylongvalueofanid",
            },
          ],
        }}
      />,
    );

    expect(element.getByText("verylong...")).toBeInTheDocument();
  });

  it("removes options when x is clicked", () => {
    // Arrange
    const element = render(
      <BulkSelect
        {...defaultProps}
        selected={{
          ...defaultProps.selected,
          value: [
            {
              label: "value1",
              value: "value1",
              slug: "value1",
            },
          ],
        }}
      />,
    );

    const option = element.getByTestId("selected-option-value1-0");
    const xButton = option.childNodes[1];

    // Act
    fireEvent.click(xButton);

    // Assert
    expect(element.queryByText("value1")).not.toBeInTheDocument();
  });

  it("removes an element when there are multiple options", () => {
    // Arrange
    const element = render(
      <BulkSelect
        {...defaultProps}
        selected={{
          ...defaultProps.selected,
          value: [
            {
              label: "value1",
              value: "value1",
              slug: "value1",
            },
            {
              label: "value2",
              value: "value2",
              slug: "value2",
            },
            {
              label: "value3",
              value: "value3",
              slug: "value3",
            },
          ],
        }}
      />,
    );

    const option = element.getByTestId("selected-option-value2-1");
    const xButton = option.childNodes[1];

    // Act
    fireEvent.click(xButton);

    // Assert
    expect(element.queryByText("value2")).not.toBeInTheDocument();
  });

  it("adds a new element when special keys are clicked", () => {
    // Arrange
    const element = render(<BulkSelect {...defaultProps} />);

    const input = element.getByTestId("right-0");

    // Act
    fireEvent.change(input, { target: { value: "value1" } });
    fireEvent.keyDown(input, { key: "," });

    fireEvent.change(input, { target: { value: "value2" } });
    fireEvent.keyDown(input, { key: " " });

    fireEvent.change(input, { target: { value: "value3" } });
    fireEvent.keyDown(input, { key: "Tab" });

    // Assert
    expect(element.getByText("value1")).toBeInTheDocument();
    expect(element.getByText("value2")).toBeInTheDocument();
    expect(element.getByText("value3")).toBeInTheDocument();
  });

  it("pastes a list of IDs and adds them as options", () => {
    // Arrange
    const element = render(<BulkSelect {...defaultProps} />);

    const input = element.getByTestId("right-0");

    // Act
    fireEvent.paste(input, { clipboardData: { getData: () => "value1, value2, value3" } });

    // Assert
    expect(element.getByText("value1")).toBeInTheDocument();
    expect(element.getByText("value2")).toBeInTheDocument();
    expect(element.getByText("value3")).toBeInTheDocument();
  });
});
