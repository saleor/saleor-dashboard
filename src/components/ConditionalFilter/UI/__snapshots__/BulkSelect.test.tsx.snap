// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConditionalFilter / UI / BulkSelect renders BulkSelect 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_18fs8ps3 _18fs8psr"
      >
        <label
          class="_18fs8ps300 _18fs8ps39o _18fs8ps1lu _18fs8ps1m0 _18fs8ps1li _18fs8ps1lo _18fs8ps1gf _18fs8ps1fi _18fs8ps1i9 _18fs8ps1hc _18fs8ps2l _18fs8ps3 _18fs8ps3do _18fs8ps250 _18fs8ps3u0 _18fs8ps3ro _18fs8ps28o _18fs8ps2ic _18fs8ps250 _18fs8ps21o _18fs8ps22d _18fs8ps10 _18fs8ps1r _18fs8psdc _18fs8psdr _18fs8ps3u0"
          data-macaw-ui-component="Multiselect"
          id="downshift-0-label"
        >
          <div
            class="_18fs8ps3 _18fs8psr _18fs8ps16u"
          >
            <span
              class="_1g8t8xp0 _9mn26xj _18fs8ps3do _18fs8ps1j3 _18fs8ps1n0 _18fs8ps1k9"
            />
            <div
              class="_18fs8ps3 _18fs8pso _18fs8psdl _18fs8ps10 _18fs8psdf"
            >
              <input
                aria-activedescendant=""
                aria-autocomplete="list"
                aria-controls="downshift-0-menu"
                aria-expanded="false"
                aria-labelledby="downshift-0-label"
                autocomplete="off"
                class="_1g8t8xp8 _18fs8ps2rc _18fs8ps310 _18fs8ps280 _18fs8ps2ho _18fs8ps1lr _18fs8ps1lx _18fs8ps1lf _18fs8ps1ll _18fs8ps16u _18fs8ps3ro _18fs8ps1j6 _18fs8ps1kc _18fs8ps1mr _18fs8ps3dc _18fs8ps3e6 _18fs8ps1j3 _18fs8ps1k9 _18fs8ps1n0 _18fs8ps14f _18fs8ps1af _18fs8ps20r"
                data-test-id="right-0"
                placeholder="Add item"
                role="combobox"
                style="flex: 1;"
                value=""
              />
            </div>
          </div>
          <div
            class="_18fs8ps14u _18fs8ps17f"
          />
        </label>
        <div
          class="_9mn26x1a _18fs8ps2s0"
        >
          <span
            class="_1ruekcr2 _1ruekcr5 _18fs8psuf _18fs8pswx _18fs8pszf _18fs8ps11x _18fs8ps3do _18fs8ps1j3 _18fs8ps1n0 _18fs8ps1k9 _18fs8ps1zo"
            data-macaw-ui-component="Text"
          >
            Helper text
          </span>
        </div>
      </div>
    </div>
    <div
      class="_8bnj8q3 _8bnj8q5 _18fs8ps2l _18fs8ps0"
      style="position: absolute; left: 0px; top: 0px; z-index: 1; pointer-events: auto; transform: translate(0px, 0px);"
    >
      <ul
        aria-labelledby="downshift-0-label"
        class="_8bnj8q8 _18fs8ps2ro _18fs8ps31c _18fs8ps28c _18fs8ps2i0 _18fs8ps1lu _18fs8ps1m0 _18fs8ps1li _18fs8ps1lo _18fs8ps2o _18fs8ps3oc _18fs8ps3v0 _18fs8ps21o _18fs8ps250 _18fs8ps16u _18fs8ps1so _18fs8ps1el _18fs8ps1y3 _18fs8ps1z0 _18fs8psuf _18fs8pswx _18fs8pszf _18fs8ps11x"
        data-macaw-ui-component="List"
        id="downshift-0-menu"
        role="listbox"
      >
        <div />
      </ul>
    </div>
  </body>,
  "container": <div>
    <div
      class="_18fs8ps3 _18fs8psr"
    >
      <label
        class="_18fs8ps300 _18fs8ps39o _18fs8ps1lu _18fs8ps1m0 _18fs8ps1li _18fs8ps1lo _18fs8ps1gf _18fs8ps1fi _18fs8ps1i9 _18fs8ps1hc _18fs8ps2l _18fs8ps3 _18fs8ps3do _18fs8ps250 _18fs8ps3u0 _18fs8ps3ro _18fs8ps28o _18fs8ps2ic _18fs8ps250 _18fs8ps21o _18fs8ps22d _18fs8ps10 _18fs8ps1r _18fs8psdc _18fs8psdr _18fs8ps3u0"
        data-macaw-ui-component="Multiselect"
        id="downshift-0-label"
      >
        <div
          class="_18fs8ps3 _18fs8psr _18fs8ps16u"
        >
          <span
            class="_1g8t8xp0 _9mn26xj _18fs8ps3do _18fs8ps1j3 _18fs8ps1n0 _18fs8ps1k9"
          />
          <div
            class="_18fs8ps3 _18fs8pso _18fs8psdl _18fs8ps10 _18fs8psdf"
          >
            <input
              aria-activedescendant=""
              aria-autocomplete="list"
              aria-controls="downshift-0-menu"
              aria-expanded="false"
              aria-labelledby="downshift-0-label"
              autocomplete="off"
              class="_1g8t8xp8 _18fs8ps2rc _18fs8ps310 _18fs8ps280 _18fs8ps2ho _18fs8ps1lr _18fs8ps1lx _18fs8ps1lf _18fs8ps1ll _18fs8ps16u _18fs8ps3ro _18fs8ps1j6 _18fs8ps1kc _18fs8ps1mr _18fs8ps3dc _18fs8ps3e6 _18fs8ps1j3 _18fs8ps1k9 _18fs8ps1n0 _18fs8ps14f _18fs8ps1af _18fs8ps20r"
              data-test-id="right-0"
              placeholder="Add item"
              role="combobox"
              style="flex: 1;"
              value=""
            />
          </div>
        </div>
        <div
          class="_18fs8ps14u _18fs8ps17f"
        />
      </label>
      <div
        class="_9mn26x1a _18fs8ps2s0"
      >
        <span
          class="_1ruekcr2 _1ruekcr5 _18fs8psuf _18fs8pswx _18fs8pszf _18fs8ps11x _18fs8ps3do _18fs8ps1j3 _18fs8ps1n0 _18fs8ps1k9 _18fs8ps1zo"
          data-macaw-ui-component="Text"
        >
          Helper text
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ConditionalFilter / UI / BulkSelect renders options inside the input 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_18fs8ps3 _18fs8psr"
      >
        <label
          class="_18fs8ps300 _18fs8ps39o _18fs8ps1lu _18fs8ps1m0 _18fs8ps1li _18fs8ps1lo _18fs8ps1gf _18fs8ps1fi _18fs8ps1i9 _18fs8ps1hc _18fs8ps2l _18fs8ps3 _18fs8ps3do _18fs8ps250 _18fs8ps3u0 _18fs8ps3ro _18fs8ps28o _18fs8ps2ic _9mn26xg _18fs8ps21o _18fs8ps22d _18fs8ps10 _18fs8ps1r _18fs8psdc _18fs8psdr _18fs8ps3u0"
          data-macaw-ui-component="Multiselect"
          id="downshift-3-label"
        >
          <div
            class="_18fs8ps3 _18fs8psr _18fs8ps16u"
          >
            <span
              class="_1g8t8xp0 _1g8t8xp2 _9mn26xj _18fs8ps3do _18fs8ps1j3 _18fs8ps1n0 _18fs8ps1k9 _9mn26xv"
            />
            <div
              class="_18fs8ps3 _18fs8pso _18fs8psdl _18fs8ps10 _18fs8psdf"
            >
              <div
                class="_18fs8ps2go _18fs8ps2qc _18fs8ps2zc _18fs8ps390 _18fs8ps1lu _18fs8ps1m0 _18fs8ps1li _18fs8ps1lo _18fs8ps1gf _18fs8ps1fi _18fs8ps1i9 _18fs8ps1hc _18fs8ps3oc _18fs8ps21o _18fs8ps250 _18fs8ps3 _18fs8psdl _18fs8ps10"
                data-test-id="selected-option-value1-0"
                tabindex="-1"
              >
                <span
                  class="_1ruekcr2 _1ruekcr5 _18fs8psuf _18fs8pswx _18fs8pszf _18fs8ps11x _18fs8ps3dc _18fs8ps1j0 _18fs8ps1mx _18fs8ps1k6 _18fs8ps1zo"
                  data-macaw-ui-component="Text"
                >
                  value1
                </span>
                <span
                  class="_1ruekcr2 _1ruekcr5 _18fs8psuf _18fs8pswx _18fs8pszf _18fs8ps11x _18fs8ps3dc _18fs8ps1j0 _18fs8ps1mx _18fs8ps1k6 _18fs8ps1zo _18fs8ps3so"
                  data-macaw-ui-component="Text"
                >
                  âœ•
                </span>
              </div>
              <input
                aria-activedescendant=""
                aria-autocomplete="list"
                aria-controls="downshift-3-menu"
                aria-expanded="false"
                aria-labelledby="downshift-3-label"
                autocomplete="off"
                class="_1g8t8xp8 _18fs8ps2rc _18fs8ps310 _18fs8ps280 _18fs8ps2ho _18fs8ps1lr _18fs8ps1lx _18fs8ps1lf _18fs8ps1ll _18fs8ps16u _18fs8ps3ro _18fs8ps1j6 _18fs8ps1kc _18fs8ps1mr _18fs8ps3dc _18fs8ps3e6 _18fs8ps1j3 _18fs8ps1k9 _18fs8ps1n0 _18fs8ps14f _18fs8ps1af _18fs8ps20r"
                data-test-id="right-0"
                placeholder="Add item"
                role="combobox"
                style="flex: 1;"
                value=""
              />
            </div>
          </div>
          <div
            class="_18fs8ps14u _18fs8ps17f"
          />
        </label>
        <div
          class="_9mn26x1a _18fs8ps2s0"
        >
          <span
            class="_1ruekcr2 _1ruekcr5 _18fs8psuf _18fs8pswx _18fs8pszf _18fs8ps11x _18fs8ps3do _18fs8ps1j3 _18fs8ps1n0 _18fs8ps1k9 _18fs8ps1zo"
            data-macaw-ui-component="Text"
          >
            Helper text
          </span>
        </div>
      </div>
    </div>
    <div
      class="_8bnj8q3 _8bnj8q5 _18fs8ps2l _18fs8ps0"
      style="position: absolute; left: 0px; top: 0px; z-index: 1; pointer-events: auto; transform: translate(0px, 0px);"
    >
      <ul
        aria-labelledby="downshift-3-label"
        class="_8bnj8q8 _18fs8ps2ro _18fs8ps31c _18fs8ps28c _18fs8ps2i0 _18fs8ps1lu _18fs8ps1m0 _18fs8ps1li _18fs8ps1lo _18fs8ps2o _18fs8ps3oc _18fs8ps3v0 _18fs8ps21o _18fs8ps250 _18fs8ps16u _18fs8ps1so _18fs8ps1el _18fs8ps1y3 _18fs8ps1z0 _18fs8psuf _18fs8pswx _18fs8pszf _18fs8ps11x"
        data-macaw-ui-component="List"
        id="downshift-3-menu"
        role="listbox"
      >
        <div />
      </ul>
    </div>
  </body>,
  "container": <div>
    <div
      class="_18fs8ps3 _18fs8psr"
    >
      <label
        class="_18fs8ps300 _18fs8ps39o _18fs8ps1lu _18fs8ps1m0 _18fs8ps1li _18fs8ps1lo _18fs8ps1gf _18fs8ps1fi _18fs8ps1i9 _18fs8ps1hc _18fs8ps2l _18fs8ps3 _18fs8ps3do _18fs8ps250 _18fs8ps3u0 _18fs8ps3ro _18fs8ps28o _18fs8ps2ic _9mn26xg _18fs8ps21o _18fs8ps22d _18fs8ps10 _18fs8ps1r _18fs8psdc _18fs8psdr _18fs8ps3u0"
        data-macaw-ui-component="Multiselect"
        id="downshift-3-label"
      >
        <div
          class="_18fs8ps3 _18fs8psr _18fs8ps16u"
        >
          <span
            class="_1g8t8xp0 _1g8t8xp2 _9mn26xj _18fs8ps3do _18fs8ps1j3 _18fs8ps1n0 _18fs8ps1k9 _9mn26xv"
          />
          <div
            class="_18fs8ps3 _18fs8pso _18fs8psdl _18fs8ps10 _18fs8psdf"
          >
            <div
              class="_18fs8ps2go _18fs8ps2qc _18fs8ps2zc _18fs8ps390 _18fs8ps1lu _18fs8ps1m0 _18fs8ps1li _18fs8ps1lo _18fs8ps1gf _18fs8ps1fi _18fs8ps1i9 _18fs8ps1hc _18fs8ps3oc _18fs8ps21o _18fs8ps250 _18fs8ps3 _18fs8psdl _18fs8ps10"
              data-test-id="selected-option-value1-0"
              tabindex="-1"
            >
              <span
                class="_1ruekcr2 _1ruekcr5 _18fs8psuf _18fs8pswx _18fs8pszf _18fs8ps11x _18fs8ps3dc _18fs8ps1j0 _18fs8ps1mx _18fs8ps1k6 _18fs8ps1zo"
                data-macaw-ui-component="Text"
              >
                value1
              </span>
              <span
                class="_1ruekcr2 _1ruekcr5 _18fs8psuf _18fs8pswx _18fs8pszf _18fs8ps11x _18fs8ps3dc _18fs8ps1j0 _18fs8ps1mx _18fs8ps1k6 _18fs8ps1zo _18fs8ps3so"
                data-macaw-ui-component="Text"
              >
                âœ•
              </span>
            </div>
            <input
              aria-activedescendant=""
              aria-autocomplete="list"
              aria-controls="downshift-3-menu"
              aria-expanded="false"
              aria-labelledby="downshift-3-label"
              autocomplete="off"
              class="_1g8t8xp8 _18fs8ps2rc _18fs8ps310 _18fs8ps280 _18fs8ps2ho _18fs8ps1lr _18fs8ps1lx _18fs8ps1lf _18fs8ps1ll _18fs8ps16u _18fs8ps3ro _18fs8ps1j6 _18fs8ps1kc _18fs8ps1mr _18fs8ps3dc _18fs8ps3e6 _18fs8ps1j3 _18fs8ps1k9 _18fs8ps1n0 _18fs8ps14f _18fs8ps1af _18fs8ps20r"
              data-test-id="right-0"
              placeholder="Add item"
              role="combobox"
              style="flex: 1;"
              value=""
            />
          </div>
        </div>
        <div
          class="_18fs8ps14u _18fs8ps17f"
        />
      </label>
      <div
        class="_9mn26x1a _18fs8ps2s0"
      >
        <span
          class="_1ruekcr2 _1ruekcr5 _18fs8psuf _18fs8pswx _18fs8pszf _18fs8ps11x _18fs8ps3do _18fs8ps1j3 _18fs8ps1n0 _18fs8ps1k9 _18fs8ps1zo"
          data-macaw-ui-component="Text"
        >
          Helper text
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ConditionalFilter / UI / BulkSelect renders options inside the input with more than 15 characters 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_18fs8ps3 _18fs8psr"
      >
        <label
          class="_18fs8ps300 _18fs8ps39o _18fs8ps1lu _18fs8ps1m0 _18fs8ps1li _18fs8ps1lo _18fs8ps1gf _18fs8ps1fi _18fs8ps1i9 _18fs8ps1hc _18fs8ps2l _18fs8ps3 _18fs8ps3do _18fs8ps250 _18fs8ps3u0 _18fs8ps3ro _18fs8ps28o _18fs8ps2ic _9mn26xg _18fs8ps21o _18fs8ps22d _18fs8ps10 _18fs8ps1r _18fs8psdc _18fs8psdr _18fs8ps3u0"
          data-macaw-ui-component="Multiselect"
          id="downshift-6-label"
        >
          <div
            class="_18fs8ps3 _18fs8psr _18fs8ps16u"
          >
            <span
              class="_1g8t8xp0 _1g8t8xp2 _9mn26xj _18fs8ps3do _18fs8ps1j3 _18fs8ps1n0 _18fs8ps1k9 _9mn26xv"
            />
            <div
              class="_18fs8ps3 _18fs8pso _18fs8psdl _18fs8ps10 _18fs8psdf"
            >
              <div
                class="_18fs8ps2go _18fs8ps2qc _18fs8ps2zc _18fs8ps390 _18fs8ps1lu _18fs8ps1m0 _18fs8ps1li _18fs8ps1lo _18fs8ps1gf _18fs8ps1fi _18fs8ps1i9 _18fs8ps1hc _18fs8ps3oc _18fs8ps21o _18fs8ps250 _18fs8ps3 _18fs8psdl _18fs8ps10"
                data-test-id="selected-option-verylongvalueofanid-0"
                tabindex="-1"
              >
                <span
                  class="_1ruekcr2 _1ruekcr5 _18fs8psuf _18fs8pswx _18fs8pszf _18fs8ps11x _18fs8ps3dc _18fs8ps1j0 _18fs8ps1mx _18fs8ps1k6 _18fs8ps1zo"
                  data-macaw-ui-component="Text"
                >
                  verylong...
                </span>
                <span
                  class="_1ruekcr2 _1ruekcr5 _18fs8psuf _18fs8pswx _18fs8pszf _18fs8ps11x _18fs8ps3dc _18fs8ps1j0 _18fs8ps1mx _18fs8ps1k6 _18fs8ps1zo _18fs8ps3so"
                  data-macaw-ui-component="Text"
                >
                  âœ•
                </span>
              </div>
              <input
                aria-activedescendant=""
                aria-autocomplete="list"
                aria-controls="downshift-6-menu"
                aria-expanded="false"
                aria-labelledby="downshift-6-label"
                autocomplete="off"
                class="_1g8t8xp8 _18fs8ps2rc _18fs8ps310 _18fs8ps280 _18fs8ps2ho _18fs8ps1lr _18fs8ps1lx _18fs8ps1lf _18fs8ps1ll _18fs8ps16u _18fs8ps3ro _18fs8ps1j6 _18fs8ps1kc _18fs8ps1mr _18fs8ps3dc _18fs8ps3e6 _18fs8ps1j3 _18fs8ps1k9 _18fs8ps1n0 _18fs8ps14f _18fs8ps1af _18fs8ps20r"
                data-test-id="right-0"
                placeholder="Add item"
                role="combobox"
                style="flex: 1;"
                value=""
              />
            </div>
          </div>
          <div
            class="_18fs8ps14u _18fs8ps17f"
          />
        </label>
        <div
          class="_9mn26x1a _18fs8ps2s0"
        >
          <span
            class="_1ruekcr2 _1ruekcr5 _18fs8psuf _18fs8pswx _18fs8pszf _18fs8ps11x _18fs8ps3do _18fs8ps1j3 _18fs8ps1n0 _18fs8ps1k9 _18fs8ps1zo"
            data-macaw-ui-component="Text"
          >
            Helper text
          </span>
        </div>
      </div>
    </div>
    <div
      class="_8bnj8q3 _8bnj8q5 _18fs8ps2l _18fs8ps0"
      style="position: absolute; left: 0px; top: 0px; z-index: 1; pointer-events: auto; transform: translate(0px, 0px);"
    >
      <ul
        aria-labelledby="downshift-6-label"
        class="_8bnj8q8 _18fs8ps2ro _18fs8ps31c _18fs8ps28c _18fs8ps2i0 _18fs8ps1lu _18fs8ps1m0 _18fs8ps1li _18fs8ps1lo _18fs8ps2o _18fs8ps3oc _18fs8ps3v0 _18fs8ps21o _18fs8ps250 _18fs8ps16u _18fs8ps1so _18fs8ps1el _18fs8ps1y3 _18fs8ps1z0 _18fs8psuf _18fs8pswx _18fs8pszf _18fs8ps11x"
        data-macaw-ui-component="List"
        id="downshift-6-menu"
        role="listbox"
      >
        <div />
      </ul>
    </div>
  </body>,
  "container": <div>
    <div
      class="_18fs8ps3 _18fs8psr"
    >
      <label
        class="_18fs8ps300 _18fs8ps39o _18fs8ps1lu _18fs8ps1m0 _18fs8ps1li _18fs8ps1lo _18fs8ps1gf _18fs8ps1fi _18fs8ps1i9 _18fs8ps1hc _18fs8ps2l _18fs8ps3 _18fs8ps3do _18fs8ps250 _18fs8ps3u0 _18fs8ps3ro _18fs8ps28o _18fs8ps2ic _9mn26xg _18fs8ps21o _18fs8ps22d _18fs8ps10 _18fs8ps1r _18fs8psdc _18fs8psdr _18fs8ps3u0"
        data-macaw-ui-component="Multiselect"
        id="downshift-6-label"
      >
        <div
          class="_18fs8ps3 _18fs8psr _18fs8ps16u"
        >
          <span
            class="_1g8t8xp0 _1g8t8xp2 _9mn26xj _18fs8ps3do _18fs8ps1j3 _18fs8ps1n0 _18fs8ps1k9 _9mn26xv"
          />
          <div
            class="_18fs8ps3 _18fs8pso _18fs8psdl _18fs8ps10 _18fs8psdf"
          >
            <div
              class="_18fs8ps2go _18fs8ps2qc _18fs8ps2zc _18fs8ps390 _18fs8ps1lu _18fs8ps1m0 _18fs8ps1li _18fs8ps1lo _18fs8ps1gf _18fs8ps1fi _18fs8ps1i9 _18fs8ps1hc _18fs8ps3oc _18fs8ps21o _18fs8ps250 _18fs8ps3 _18fs8psdl _18fs8ps10"
              data-test-id="selected-option-verylongvalueofanid-0"
              tabindex="-1"
            >
              <span
                class="_1ruekcr2 _1ruekcr5 _18fs8psuf _18fs8pswx _18fs8pszf _18fs8ps11x _18fs8ps3dc _18fs8ps1j0 _18fs8ps1mx _18fs8ps1k6 _18fs8ps1zo"
                data-macaw-ui-component="Text"
              >
                verylong...
              </span>
              <span
                class="_1ruekcr2 _1ruekcr5 _18fs8psuf _18fs8pswx _18fs8pszf _18fs8ps11x _18fs8ps3dc _18fs8ps1j0 _18fs8ps1mx _18fs8ps1k6 _18fs8ps1zo _18fs8ps3so"
                data-macaw-ui-component="Text"
              >
                âœ•
              </span>
            </div>
            <input
              aria-activedescendant=""
              aria-autocomplete="list"
              aria-controls="downshift-6-menu"
              aria-expanded="false"
              aria-labelledby="downshift-6-label"
              autocomplete="off"
              class="_1g8t8xp8 _18fs8ps2rc _18fs8ps310 _18fs8ps280 _18fs8ps2ho _18fs8ps1lr _18fs8ps1lx _18fs8ps1lf _18fs8ps1ll _18fs8ps16u _18fs8ps3ro _18fs8ps1j6 _18fs8ps1kc _18fs8ps1mr _18fs8ps3dc _18fs8ps3e6 _18fs8ps1j3 _18fs8ps1k9 _18fs8ps1n0 _18fs8ps14f _18fs8ps1af _18fs8ps20r"
              data-test-id="right-0"
              placeholder="Add item"
              role="combobox"
              style="flex: 1;"
              value=""
            />
          </div>
        </div>
        <div
          class="_18fs8ps14u _18fs8ps17f"
        />
      </label>
      <div
        class="_9mn26x1a _18fs8ps2s0"
      >
        <span
          class="_1ruekcr2 _1ruekcr5 _18fs8psuf _18fs8pswx _18fs8pszf _18fs8ps11x _18fs8ps3do _18fs8ps1j3 _18fs8ps1n0 _18fs8ps1k9 _18fs8ps1zo"
          data-macaw-ui-component="Text"
        >
          Helper text
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
